taxonomy/migrations_utils.py

```diff
@@ -0,0 +1,70 @@
+"""
+Utils used in migrations.
+"""
+from django.db.models import Count
+
+
+def remove_all_jobs(apps, schema_editor):  # pylint: disable=unused-argument
+    """
+    Delete all Jobs object as we are adding a unique non-nullable `external_id`, that we cannot back-populate.
+    """
+    job_model = apps.get_model('taxonomy', 'Job')
+    job_model.objects.all().delete()
+
+
+def remove_duplicates_skills(apps, schema_editor):  # pylint: disable=unused-argument
+    """
+    Remove all duplicate records from the Skill table.
+    """
+    skill_table = apps.get_model('taxonomy', 'Skill')
+    duplicate_skills = skill_table.objects.values('external_id') \
+        .annotate(skill_count=Count('id')).order_by() \
+        .filter(skill_count__gt=1)
+    for skill in duplicate_skills:
+        skill_ids = list(skill_table.objects.filter(external_id=skill['external_id']).values_list('id', flat=True))[1:]
+        skill_table.objects.filter(pk__in=skill_ids).delete()
+
+
+def remove_duplicates_course_skills(apps, schema_editor):  # pylint: disable=unused-argument
+    """
+    Remove all duplicate records having same course_id and skill_id from the CourseSkills table.
+    """
+    course_skill_table = apps.get_model('taxonomy', 'CourseSkills')
+    duplicate_course_skills = course_skill_table.objects.values('course_id', 'skill_id') \
+        .annotate(course_skill_count=Count('id')).order_by() \
+        .filter(course_skill_count__gt=1)
+    for courseskill in duplicate_course_skills:
+        skill_ids = list(course_skill_table.objects.filter(
+            job_id=courseskill['course_id'], skill_id=courseskill['skill_id']
+        ).values_list('id', flat=True))[1:]
+        course_skill_table.objects.filter(pk__in=skill_ids).delete()
+
+
+def remove_duplicates(apps, schema_editor):  # pylint: disable=unused-argument
+    """
+    Remove all duplicate records having same course_id and skill_id from the JobPostings table.
+    """
+    job_postings_table = apps.get_model('taxonomy', 'JobPostings')
+    duplicate_job_postings = job_postings_table.objects.values('job_id') \
+        .annotate(job_postings_count=Count('id')).order_by() \
+        .filter(job_postings_count__gt=1)
+    for jobpostings in duplicate_job_postings:
+        job_postings_ids = list(job_postings_table.objects.filter(
+            job_id=jobpostings['job_id']
+        ).values_list('id', flat=True))[1:]
+        job_postings_table.objects.filter(pk__in=job_postings_ids).delete()
+
+
+def remove_duplicates_job_skills(apps, schema_editor):  # pylint: disable=unused-argument
```

Comment:
Why are we passing `schema_editor` as argument if we are not using it?

Reply:
These methods will be called from the `migrations.RunPython` that expects these params.